### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by indraraj**
- Updated the folder structure to add `connectorStepHelpers` inside /components/ and `connectorSteps` to only have steps component.
- Uplifted the connector mock to the Host app and using the connector data geting via `connector` props.
- Updating the Properties name by replacing `.` with `_` in DebeziumConfigurator `connectorProperties` state itself.
- Fixed the `configuration` value passed in `onChange` in DataOption and RuntimeOption component initilization. 
- Converted the variable(like `snapshotPropertyDefinitions`, `heartbeatPropertyDefinitions`) used for placement of properties in diff section to React state and lazy initialize the state value for performance optimization.
- Removed the deprecated ComponsntWillReciveProps from `WithLoader` and `WithErrorBoundry`.
- Updated the `WillErrorBoundry` and linked the Jira Issue link(`https://issues.redhat.com/browse/DBZ`) to "Report this issue" button.
- Updated the React and React-dom version to `17.0.1`.
- Updated the UI readme.md

### CODE REVIEW

File: ui/packages/ui/src/app/pages/createConnector/federatedModule/DataOptions.tsx [hunk 1]
```diff
@@ -39,35 +39,48 @@ const checkIfRequired = (
...
+    (defn: any) => defn.category === PropertyCategory.DATA_OPTIONS_GENERAL
+  );
+};
+
+const getMappingAdvanceProperty = (
+  propertyList: ConnectorProperty[]
+): ConnectorProperty[] => {
+  const propertyDefinitionsCopy = _.cloneDeep(propertyList);
+  return propertyDefinitionsCopy.filter(
+    (defn: any) => defn.category === PropertyCategory.DATA_OPTIONS_GENERAL
```
Comment (mdrillin) [inline ui/packages/ui/src/app/pages/createConnector/federatedModule/DataOptions.tsx:61]: This one should be PropertyCategory.DATA_OPTIONS_ADVANCED

**Review by mdrillin — APPROVED** [review]
Looks good - just one comment where there is a wrong category.  If you want to fix that one and merge, I'm good with it.

**Review by uidoyen — APPROVED** [review]
Tested locally. Looks good apart from @mdrillin comment.

File: ui/packages/ui/src/app/pages/createConnector/federatedModule/DataOptions.tsx (hunk 1 above)
Comment (indraraj) [reply ui/packages/ui/src/app/pages/createConnector/federatedModule/DataOptions.tsx:61]: done.

**Pull request merged**